Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/BasketViewController.m

```diff
@@ -54,6 +55,7 @@ - (IBAction)donateButtonTapped:(id)sender {
 
 - (void)favoriteUnfavoriteBasket {
     NSMutableArray<Basket*> *favBaskets = [[User currentUser] favoriteBaskets];
+    [self.basket fetch];
```

Comment:
If this `fetch` method is synchronous you shouldn't be calling it on the main thread, you should either move it to a background thread or use an asynchronous variant with a completion block.